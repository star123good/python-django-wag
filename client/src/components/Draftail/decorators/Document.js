import PropTypes from 'prop-types';
import React from 'react';

import Icon from '../../Icon/Icon';

import TooltipEntity from '../decorators/TooltipEntity';

const Document = props => {
  const { entityKey, contentState } = props;
  const { url } = contentState.getEntity(entityKey).getData();
  const filename = url.split('/')[3];

  return (
    <TooltipEntity
      {...props}
      icon={<Icon name="doc-full" />}
      label={filename}
      url={url}
    />
  );
};

Document.propTypes = {
  entityKey: PropTypes.string.isRequired,
  contentState: PropTypes.object.isRequired,
};

export default Document;
