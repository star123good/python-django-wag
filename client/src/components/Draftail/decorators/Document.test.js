import React from 'react';
import { shallow } from 'enzyme';
import { convertFromHTML, ContentState } from 'draft-js';
import Document from './Document';

describe('Document', () => {
  it('exists', () => {
    expect(Document).toBeDefined();
  });

  it('renders', () => {
    const contentBlocks = convertFromHTML('<h1>aaaaaaaaaa</h1>');
    const contentState = ContentState.createFromBlockArray(contentBlocks);
    const contentStateWithEntity = contentState.createEntity('DOCUMENT', 'MUTABLE', { title: 'Test title' });
    const entityKey = contentStateWithEntity.getLastCreatedEntityKey();
    expect(shallow((
      <Document
        entityKey={entityKey}
        contentState={contentStateWithEntity}
      >
        <span>Test children</span>
      </Document>
    ))).toMatchSnapshot();
  });
});
