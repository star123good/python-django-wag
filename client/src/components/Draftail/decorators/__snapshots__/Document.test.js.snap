// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document no data 1`] = `
<TooltipEntity
  contentState={
    Immutable.Record {
      "entityMap": Object {
        "__add": [Function],
        "__create": [Function],
        "__get": [Function],
        "__getLastCreatedEntityKey": [Function],
        "__mergeData": [Function],
        "__replaceData": [Function],
        "add": [Function],
        "create": [Function],
        "get": [Function],
        "getLastCreatedEntityKey": [Function],
        "mergeData": [Function],
        "replaceData": [Function],
      },
      "blockMap": Immutable.OrderedMap {
        "a": Immutable.Record {
          "key": "a",
          "type": "unstyled",
          "text": "test",
          "characterList": Immutable.List [
            Immutable.Record {
              "style": Immutable.OrderedSet [],
              "entity": null,
            },
            Immutable.Record {
              "style": Immutable.OrderedSet [],
              "entity": null,
            },
            Immutable.Record {
              "style": Immutable.OrderedSet [],
              "entity": null,
            },
            Immutable.Record {
              "style": Immutable.OrderedSet [],
              "entity": null,
            },
          ],
          "depth": 0,
          "data": Immutable.Map {},
        },
      },
      "selectionBefore": Immutable.Record {
        "anchorKey": "a",
        "anchorOffset": 0,
        "focusKey": "a",
        "focusOffset": 0,
        "isBackward": false,
        "hasFocus": false,
      },
      "selectionAfter": Immutable.Record {
        "anchorKey": "a",
        "anchorOffset": 0,
        "focusKey": "a",
        "focusOffset": 0,
        "isBackward": false,
        "hasFocus": false,
      },
    }
  }
  entityKey="2"
  icon={
    <Icon
      className={null}
      name="doc-full"
      title={null}
    />
  }
  label=""
  onEdit={[Function]}
  onRemove={[Function]}
  url=""
>
  test
</TooltipEntity>
`;

exports[`Document works 1`] = `
<TooltipEntity
  contentState={
    Immutable.Record {
      "entityMap": Object {
        "__add": [Function],
        "__create": [Function],
        "__get": [Function],
        "__getLastCreatedEntityKey": [Function],
        "__mergeData": [Function],
        "__replaceData": [Function],
        "add": [Function],
        "create": [Function],
        "get": [Function],
        "getLastCreatedEntityKey": [Function],
        "mergeData": [Function],
        "replaceData": [Function],
      },
      "blockMap": Immutable.OrderedMap {
        "a": Immutable.Record {
          "key": "a",
          "type": "unstyled",
          "text": "test",
          "characterList": Immutable.List [
            Immutable.Record {
              "style": Immutable.OrderedSet [],
              "entity": null,
            },
            Immutable.Record {
              "style": Immutable.OrderedSet [],
              "entity": null,
            },
            Immutable.Record {
              "style": Immutable.OrderedSet [],
              "entity": null,
            },
            Immutable.Record {
              "style": Immutable.OrderedSet [],
              "entity": null,
            },
          ],
          "depth": 0,
          "data": Immutable.Map {},
        },
      },
      "selectionBefore": Immutable.Record {
        "anchorKey": "a",
        "anchorOffset": 0,
        "focusKey": "a",
        "focusOffset": 0,
        "isBackward": false,
        "hasFocus": false,
      },
      "selectionAfter": Immutable.Record {
        "anchorKey": "a",
        "anchorOffset": 0,
        "focusKey": "a",
        "focusOffset": 0,
        "isBackward": false,
        "hasFocus": false,
      },
    }
  }
  entityKey="1"
  icon={
    <Icon
      className={null}
      name="doc-full"
      title={null}
    />
  }
  label="example.pdf"
  onEdit={[Function]}
  onRemove={[Function]}
  url="/example.pdf"
>
  test
</TooltipEntity>
`;
